composer/utils/inference.py

```diff
@@ -130,15 +153,11 @@ def export_for_inference(
     # Move model and sample input to CPU for export
     cpu = torch.device('cpu')
     model.to(device=cpu)
+
     if sample_input is not None:
         sample_input = ensure_tuple(sample_input)
-        for i in range(len(sample_input)):
-            if isinstance(sample_input[i], torch.Tensor):
-                sample_input[i] = sample_input[i].to(cpu)  # type: ignore
-            elif isinstance(sample_input[i], dict):
-                for key, value in sample_input[i].items():
-                    if isinstance(value, torch.Tensor):
-                        sample_input[i][key] = value.to(cpu)
+        sample_input = _move_sample_input_to_device(sample_input, cpu)
+        assert isinstance(sample_input, tuple)
```

Comment:
Asserts are helpful for tests, less useful for code. Remove since this should always be true given the ensure_tuple above and code in _move_sample_to_device.

Reply:
Ah sorry, I forgot to leave a comment saying this is for pyright. Will add

Reply:
Ok, I failed at getting pyright to accept my use of generics to try to type this function better, but I moved the assert to where the pyright error comes from and added a comment.